[DeepSLAM-2] implement enc-dec layer2

bellatoris opened the pull request:
This pr implements DeMoNs's second enc-dec layer which predicts motion.

dongwoohhh commented:
1. Where is forward function for conv block?

2. Does 1-d Kernel design(n x 1, 1 x n kernels) included?
  - Maybe no and not important.

3. layer5 - fist convBlock's kerenl size : 5 -> 3

4. motion_prediction(4 x 16 x 12) input should be considered in layer7 (516 channel inputs)

bellatoris commented:
I addressed your comments about number 3.   
And you can find the `ConvBlock`'s forward function head of the file.  
And I include the 1-d direction convolution filter, too.  
But I don't implement the multiplying scaling factor yet, we need to discuss about it later

bellatoris commented on code/DeMoN_py/encdec.py:
```diff
@@ -0,0 +1,244 @@
+        self.conv2 = nn.Conv2d(output_plane, output_plane, kernel_size=(1, kernel),
+                               stride=(1, stride), padding=(0, pad))
+
+    def forward(self, x):
```
Here it is!
